### ISSUE TITLE
Request: C source code

### ISSUE THREAD

**Issue by maxton** [sleirsgoevy/ps4jb#3]
Would it be possible to make the C sources available so we can modify and re-compile the ROP chains?

**Comment by sleirsgoevy** [comment]
Yes, I will upload the source in the next few days. The issue here is that it needs to be compiled on a FreeBSD 9 system to use the right headers, and setting up the toolchain on that is not a breeze. So I will upload it as soon as building on Linux is possible.

**Comment by maxton** [comment]
Awesome, thank you!

**Comment by eduardo1954** [comment]
Early versions of the toolchain have bsd headers
v.02

Specterdev said

**Comment by StartForKillerMC** [comment]
Maybe you can undefine and redefine the bad headers

**Comment by eduardo1954** [comment]
https://github.com/OpenOrbis/OpenOrbis-PS4-Toolchain/releases/tag/v0.2

**Comment by StartForKillerMC** [comment]
The c source code will help to solve the stability issues

**Comment by sleirsgoevy** [comment]
Note: built-from-source files differ from the bundled ones for some reason. Haven't tested them yet as I don't have access to PS4 right now. If somebody confirms that it works I'll close this.

**Comment by StartForKillerMC** [comment]
I will test it now

**Comment by StartForKillerMC** [comment]
Okay, at least for me isn't working, always says reboot you ps4

**Comment by StartForKillerMC** [comment]
It seems that it don't execute all the write_mem, i check it with some prints

**Comment by sleirsgoevy** [comment]
That is very strange. `write_mem` is implemented directly in JS, so it shouldn't break.

**Comment by StartForKillerMC** [comment]
Yeah, i know, but i tested it and it isn't working

**Comment by sleirsgoevy** [comment]
Seems that the linux build can't win the race for some reason. Investigating...

**Comment by StartForKillerMC** [comment]
I tried again and it's the same

**Comment by sleirsgoevy** [comment]
Will probably try to reproduce exactly this when I get my hands on WSL. At least the build process now works properly for me.

**Comment by ryandeanxro** [comment]
the problem you have with compiled ROP code is mainly two things for one, you have a lot of memory pressure and noise with how much more stuff is going on the scheduler is doing more work which will introduce noise in your race the other thing is it's likely it'll produce more complex / longer chains than hand-written ones will, more instructions = more execution time = more potential to lose the race 

Said By : SPECTER DEV

**Comment by StartForKillerMC** [comment]
Yeah, that's true. The main thing here it's the payload compiled doesn't work like others compiled.

**Comment by BillaBong1KP** [comment]
So the problem is MIRA or other payload ?

**Comment by StartForKillerMC** [comment]
The jailbreak, if you read the write_mem isn't executed

**Comment by BillaBong1KP** [comment]
How they will fix it ? Damm sure something is wrong with compiling the the source to rop thats why it is not executed.

**Comment by StartForKillerMC** [comment]
At least I'm sure that hosted version on my pc works, so it's not my server

**Comment by StartForKillerMC** [comment]
Maybe it's a wsl problem

**Comment by BillaBong1KP** [comment]
Yeah

**Comment by StartForKillerMC** [comment]
![image](https://user-images.githubusercontent.com/21644562/87981163-0c370a80-cad5-11ea-8200-feb21b716a11.png)
Interesting

**Comment by StartForKillerMC** [comment]
some changes from source

**Comment by BillaBong1KP** [comment]
Hope this make the exploit stable 🤟🏻

**Comment by sleirsgoevy** [comment]
> the problem you have with compiled ROP code is mainly two things for one, you have a lot of memory pressure and noise with how much more stuff is going on the scheduler is doing more work which will introduce noise in your race the other thing is it's likely it'll produce more complex / longer chains than hand-written ones will, more instructions = more execution time = more potential to lose the race
> 
> Said By : SPECTER DEV

That is not **the** problem. Failing a race could potentially result in a hangup, but not in a kernel panic.

**Comment by BillaBong1KP** [comment]
Hope you could fix this soon ASAP 👍

**Comment by sleirsgoevy** [comment]
I forgot to mention that an issue was fixed in ps4-rop-8cc (loaded into src/8cc as a submodule), so you may need to run `git pull` there to make sure that you have that patch.

**Comment by mtnjustme** [comment]
> 
> 
> I forgot to mention that an issue was fixed in ps4-rop-8cc (loaded into src/8cc as a submodule), so you may need to run `git pull` there to make sure that you have that patch.

some say that the new changes make it laggy, though it might be mira, up for your consideration and testings

**Comment by mtnjustme** [comment]
also i'd suggest looking at darkmodder's work, he added 3 alerts in the jailbreak js, sounds silly but many claim it works better.
watch this :
http://prntscr.com/tlhx9w

**Comment by BillaBong1KP** [comment]
For 8/10 people its crashing while running the jailbreak process or after completing .

**Comment by mtnjustme** [comment]
> eople its crashing while running the jailbreak proce

still worth exploring in my opinion...

**Comment by StartForKillerMC** [comment]
> I forgot to mention that an issue was fixed in ps4-rop-8cc (loaded into src/8cc as a submodule), so you may need to run `git pull` there to make sure that you have that patch.

Yeah, i know, i tried both

**Comment by StartForKillerMC** [comment]
I've tested to build it with arch linux. The only change i'm seeing:

![image](https://user-images.githubusercontent.com/21644562/88041912-64105880-cb4b-11ea-81a5-1daeb16ac690.png)
Seems that sometimes adds innecesary moves

**Comment by gotwig** [comment]
did you try to "fix" those moves?

**Comment by StartForKillerMC** [comment]
Nope, i think it's a problem related with wsl2, i need to try again with ubuntu
